libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -231,51 +231,91 @@ end
+                                             endpoint_LB[2]+ans_LB[2]+gait_pos_y[3], 
+                                             endpoint_LB[3]+ans_LB[3]+gait_pos_z[3])
+    angles_LB[1] = -45 + angles_LB[1]
+
+    local ans_LF = body_forward_kinematics(endpoint_LF[1]+gait_pos_x[4],
+                                           endpoint_LF[2]+gait_pos_y[4], 
+                                           endpoint_LF[3]+gait_pos_z[4], 
+                                           FRAME_LEN*0.5, -FRAME_WIDTH*0.5, 
+                                           gait_rot_z[4])
+    local angles_LF = leg_inverse_kinematics(endpoint_LF[1]-ans_LF[1]+gait_pos_x[4],
+                                             endpoint_LF[2]-ans_LF[2]-gait_pos_y[4], 
+                                             endpoint_LF[3]+ans_LF[3]+gait_pos_z[4])
+    angles_LF[1] = -135 + angles_LF[1]
     Gaitselect()
-    current = {angles1[1],angles1[2],angles1[3],angles2[1],angles2[2],angles2[3],angles3[1],angles3[2],angles3[3],angles4[1],angles4[2],angles4[3]}
+    local current_angle = {angles_RF[1],angles_RF[2],angles_RF[3],
+                           angles_LF[1],angles_LF[2],angles_LF[3],
+                           angles_LB[1],angles_LB[2],angles_LB[3],
+                           angles_RB[1],angles_RB[2],angles_RB[3]}
 
-    if servo_estimate() then
+    if servo_estimate(current_angle) then
         start_time = millis()
         calc_gait_sequence()
-        last_angle = current
+        last_angle = current_angle
     end
 
-    return current
+    return current_angle
 end
 
 -- servo angles when robot is disarmed and resting body on the ground
 local rest_angles = { 45, -90, 40,      -- front right leg (coxa, femur, tibia)
                      -45, -90, 40,      -- front left leg (coxa, femur, tibia)
-                      45, -90, 40,      -- back right leg (coxa, femur, tibia)
-                     -45, -90, 40}      -- back left leg (coxa, femur, tibia)
-
-local servo_direction = { 1,  1,  1,    -- front right leg (coxa, femur, tibia)
-                         -1, -1, -1,    -- front left leg (coxa, femur, tibia)
-                         -1, -1,  1,    -- back right leg (coxa, femur, tibia)
-                          1,  1, -1}    -- back left leg (coxa, femur, tibia)
+                     -45, -90, 40,      -- back left leg (coxa, femur, tibia)
+                      45, -90, 40}      -- back right leg (coxa, femur, tibia)
 
 function update()
-    x_speed = vehicle:get_control_output(control_input_throttle) * xy_speed_max
-    yaw_speed = vehicle:get_control_output(control_input_yaw) * yaw_speed_max
+    local throttle = vehicle:get_control_output(control_input_throttle) * xy_travel_max
+    local gait_direction
+    if throttle > 0 then
+        gait_direction = -1
+        y_travel = throttle
+    elseif throttle < 0 then
+        gait_direction = 1
+        y_travel = - throttle
+    elseif throttle == 0 then
+        gait_direction = 1
+        y_travel = 0
```

Comment:
Does this mean the robot can't rotate if it is not moving forward or backwards?

Reply:
the robot can rotate without throttle , the yaw has no relation with the throttle.

Reply:
Ah, right.. I was thinking "yaw_travel" instead of "y_travel"